doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
-   inductive or co-inductive type. Unlike :tacn:`induction`, no induction
-   hypothesis is generated by :tacn:`destruct`.
+   .. coqtop:: reset none
 
-   .. tacv:: destruct @ident
+      Goal  forall P1 P2 : Prop, P1 -> P1 \/ P2.
 
-      If :token:`ident` denotes a quantified variable of the conclusion
-      of the goal, then :n:`destruct @ident` behaves
-      as :n:`intros until @ident; destruct @ident`. If :token:`ident` is not
-      anymore dependent in the goal after application of :tacn:`destruct`, it
-      is erased (to avoid erasure, use parentheses, as in :n:`destruct (@ident)`).
+   .. coqtop:: all
 
-      If :token:`ident` is a hypothesis of the context, and :token:`ident`
-      is not anymore dependent in the goal after application
-      of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
-      in :n:`destruct (@ident)`).
+      constructor 2.  (* equivalent to "right" *)
 
-   .. tacv:: destruct @natural
+   You can also apply a constructor by name.  (`apply or_intror` will fail):
 
-      :n:`destruct @natural` behaves as :n:`intros until @natural`
-      followed by destruct applied to the last introduced hypothesis.
+   .. coqtop:: reset none
 
-     .. note::
-        For destruction of a number, use syntax :n:`destruct (@natural)` (not
-        very interesting anyway).
+      Goal  forall P1 P2 : Prop, P1 -> P1 \/ P2.
 
-   .. tacv:: destruct @pattern
+   .. coqtop:: all
 
-      The argument of :tacn:`destruct` can also be a pattern of which holes are
-      denoted by “_”. In this case, the tactic checks that all subterms
-      matching the pattern in the conclusion and the hypotheses are compatible
-      and performs case analysis using this subterm.
+      apply or_introl.  (* equivalent to "left" *)
 
-   .. tacv:: destruct {+, @term}
 
-      This is a shortcut for :n:`destruct @term; ...; destruct @term`.
+.. _CaseAnalysisAndInduction:
 
-   .. tacv:: destruct @term as @or_and_intropattern_loc
+Case analysis
+-------------
 
-      This behaves as :n:`destruct @term` but uses the names
-      in :token:`or_and_intropattern_loc` to name the variables introduced in the
-      context. The :token:`or_and_intropattern_loc` must have the
-      form :n:`[p11 ... p1n | ... | pm1 ... pmn ]` with ``m`` being the
-      number of constructors of the type of :token:`term`. Each variable
-      introduced by :tacn:`destruct` in the context of the ``i``-th goal
-      gets its name from the list :n:`pi1 ... pin` in order. If there are not
-      enough names, :tacn:`destruct` invents names for the remaining variables
-      to introduce. More generally, the :n:`pij` can be any introduction
-      pattern (see :tacn:`intros`). This provides a concise notation for
-      chaining destruction of a hypothesis.
+The tactics in this section implement case
+analysis on inductive or co-inductive objects (see :ref:`gallina-inductive-definitions`).
```

Comment:
In fact, the most relevant section to refer to at this point is "Variants and the `match` construct" (since this is the section where recursion doesn't matter).